packages/blade/src/components/InputGroup/_decisions/decisions.md

```diff
@@ -0,0 +1,165 @@
+
+### Prop Table
+
+| Prop            | Type                           | Default  | Description                                                     |
+| --------------- | ------------------------------ | -------- | --------------------------------------------------------------- |
+| label           | string                         | -        | Label for the entire input group.                               |
+| labelPosition   | 'top' \| 'left'                | 'top'    | Position of the label relative to the group.                    |
+| size            | 'medium' \| 'large'            | 'medium' | Controls the size of the input group and its child inputs.      |
+| helpText        | string                         | -        | Help text displayed at the bottom of the group.                 |
+| validationState | 'error' \| 'success' \| 'none' | 'none'   | Controls which validation message is displayed.                 |
+| errorText       | string                         | -        | Error message that appears when validationState is 'error'.     |
+| successText     | string                         | -        | Success message that appears when validationState is 'success'. |
+| isDisabled      | boolean                        | false    | Disables all inputs within the group.                           |
+| children        | React.ReactNode                | -        | Should be InputRow components or other valid inputs.            |
+| testID          | string                         | -        | Test ID for automation.                                         |
+
+## `InputRow` API
+
+### Prop Table
+
+| Prop            | Type            | Default | Description                                                                                                               |
+| --------------- | --------------- | ------- | ------------------------------------------------------------------------------------------------------------------------- |
+| templateColumns | string          | -       | CSS grid template columns value (e.g., "1fr 2fr" or "200px 1fr"). Controls how space is distributed between child inputs. |
+| children        | React.ReactNode | -       | Input components to render in this row.                                                                                   |
+| testID          | string          | -       | Test ID for automation.                                                                                                   |
+
+## Usage
+
+### Basic Usage
+
+```jsx
+import { InputGroup, InputRow, TextInput, PasswordInput } from '@razorpay/blade/components';
+
+const App = () => {
+  return (
+    <InputGroup label="Card Details" helpText="Enter details about the card">
+      <InputRow>
+        <TextInput placeholder="4400-1234-5422-6574" />
+      </InputRow>
+      <InputRow templateColumns="1fr 2fr">
+        <TextInput placeholder="John Doe" />
+        <PasswordInput placeholder="•••••" />
```

Comment:
how can we handle responsiveness?

Like can I make TextInput and PasswordInput be on different rows on mobile in this case?

Reply:
Makes sense